lib/state_change_syncer.go

```diff
@@ -0,0 +1,867 @@
+			if err != nil {
+				fmt.Printf("StateChangeSyncer.StartMempoolSyncRoutine: Error flushing all entries to file: %v", err)
+			}
+		}
+		mempoolClosed := server.mempool.stopped
+		for !mempoolClosed {
+			// Sleep for a short while to avoid a tight loop.
+			time.Sleep(100 * time.Millisecond)
+			var err error
+			// If the mempool is not empty, sync the mempool to the state syncer.
+			mempoolClosed, err = stateChangeSyncer.SyncMempoolToStateSyncer(server)
+			if err != nil {
+				glog.Errorf("StateChangeSyncer.StartMempoolSyncRoutine: Error syncing mempool to state syncer: %v", err)
+			}
+		}
+	}()
+}
+
+func (stateChangeSyncer *StateChangeSyncer) FlushAllEntriesToFile(server *Server) error {
+	// Check if the state change file already exists and is not empty. If so, return.
+	stateChangeFileInfo, err := stateChangeSyncer.StateChangeFile.Stat()
+	if err == nil {
+		// If the file is non-empty, no need to flush entries to file.
+		if stateChangeFileInfo.Size() > 0 {
+			return nil
+		}
+	}
+
+	// Disable deadlock detection, as the process of flushing entries to file can take a long time and
+	// if it takes longer than the deadlock detection timeout interval, it will cause an error to be thrown.
+	deadlock.Opts.Disable = true
+	defer func() {
+		deadlock.Opts.Disable = false
+	}()
+	// Lock the blockchain so that nothing shifts under our feet while dumping the current state to the state change file.
+	server.blockchain.ChainLock.Lock()
+	defer server.blockchain.ChainLock.Unlock()
+
+	fmt.Printf("Flushing all block-synced entries to state change file.\n")
+	// Allow the state change syncer to flush entries to file.
+	stateChangeSyncer.BlocksyncCompleteEntriesFlushed = true
+
+	// Loop through all prefixes that hold state change entries.
+	for _, prefix := range StatePrefixes.CoreStatePrefixesList {
+		// Start with the first key in the prefix.
+		lastReceivedKey := prefix
+		chunkFull := true
+		var err error
+		var dbBatchEntries []*DBEntry
+
+		// Loop through all the batches of entries for the prefix until we get a non-full chunk.
+		for chunkFull {
+			fmt.Printf("Processing chunk for prefix: %+v\n", prefix)
+			// Create a flush ID for this chunk.
+			dbFlushId := uuid.New()
+			// Fetch the batch from main DB records with a batch size of about snap.BatchSize.
+			dbBatchEntries, chunkFull, err = DBIteratePrefixKeys(server.blockchain.db, prefix, lastReceivedKey, SnapshotBatchSize/10)
+			fmt.Printf("Got %d entries for prefix: %+v\n", len(dbBatchEntries), prefix)
+			if err != nil {
+				return errors.Wrapf(err, "StateChangeSyncer.FlushAllEntriesToFile: ")
+			}
+			if len(dbBatchEntries) != 0 {
+				lastReceivedKey = dbBatchEntries[len(dbBatchEntries)-1].Key
+			}
+			for ii, dbEntry := range dbBatchEntries {
+				stateChangeEntry := &StateChangeEntry{
+					OperationType: DbOperationTypeInsert,
+					KeyBytes:      dbEntry.Key,
+					EncoderBytes:  dbEntry.Value,
+				}
+
+				fmt.Printf("Flushing entry %d to file\n", ii)
```

Comment:
Do you want to leave all these prints in here? Fine if they're intentional I'm just flagging it.